source/docs/software/hardware-apis/sensors/gyros-software.rst

```diff
@@ -52,10 +52,43 @@ The :code:`AnalogGyro` class (`Java <https://first.wpi.edu/wpilib/allwpilib/docs
         // Creates an AnalogGyro object on port 0
         frc::AnalogGyro gyro{0};
 
-Third-party gyros
------------------
+navX
+^^^^
 
-While WPILib provides native support for a the ADXRS450 gyro available in the kit of parts and for any analog gyro, there are a few popular AHRS (Attitude and Heading Reference System) devices commonly used in FRC\ |reg| that include accelerometers and require more complicated communications.  These are generally controlled through vendor libraries.
+The navX uses the :code:`AHRS` class and implements the :code:`Gyro` interface.  See the `navX documentation <https://pdocs.kauailabs.com/navx-mxp/guidance/selecting-an-interface/>`__ for additional connection types.
+
+.. tabs::
+
+    .. code-tab:: java
+
+        // navX MXP using SPI
+        AHRS gyro = new AHRS(SPI.Port.kMXP);
+
+    .. code-tab:: c++
+
+        // navX MXP using SPI
+        AHRS gyro{SPI::Port::kMXP};
+
+Pigeon
+^^^^^^
+
+The pigeon should use the :code:`WPI_PigeonIMU` class that implements :code:`Gyro`.  The Pigeon can either be connected with CAN or by data cable to a TalonSRX.  The `Pigeon IMU User's Guide <https://www.ctr-electronics.com/downloads/pdf/Pigeon%20IMU%20User%27s%20Guide.pdf>`__ contains full details on using the Pigeon.
+
+.. tabs::
+
+    .. code-tab:: java
+
+        WPI_PigeonIMU gyro = new WPI_PigeonIMU(0); // Pigeon is on CAN Bus with device ID 0
```

Comment:
I don't see WPI_PigeonIMU in the api docs for Phoenix.

Reply:
1) Yes CAN ID 0 can be Pigeon or Talon or even both since they are separate devices.
2) I am using it in some other code I am writing so maybe it is a beta thing?  @JCaporuscio will this be in the docs on release?

Reply:
I don't think our API docs have been updated since the last official public release and WPI_PiggeonIMU was added after that.
So it should get picked up when we generate the API docs for kickoff.